tsdb/head_append.go

```diff
@@ -290,15 +290,22 @@ func (s *memSeries) appendable(t int64, v float64) error {
 	if t > c.maxTime {
 		return nil
 	}
-	if t < c.maxTime {
-		return storage.ErrOutOfOrderSample
-	}
+
 	// We are allowing exact duplicates as we can encounter them in valid cases
 	// like federation and erroring out at that time would be extremely noisy.
-	if math.Float64bits(s.sampleBuf[3].v) != math.Float64bits(v) {
+	if t == c.maxTime && math.Float64bits(s.sampleBuf[3].v) != math.Float64bits(v) {
 		return storage.ErrDuplicateSampleForTimestamp
```

Comment:
I think we don't intend to send `ErrDuplicateSampleForTimestamp` at all now. Can you confirm @bboreham? But I wonder if it will hide any bugs in metric exposition.

Reply:
No, that was not what I wanted.
Purely to allow duplicates of the last 4, as an improvement on the last 1.